pkg/watermark/fetch/processor_manager.go

```diff
@@ -212,3 +212,52 @@ func (v *ProcessorManager) startHeatBeatWatcher() {
 
 	}
 }
+
+func (v *ProcessorManager) startTimeLineWatcher() {
+	watchCh, stopped := v.otWatcher.Watch(v.ctx)
+
+	for {
+		select {
+		case <-stopped:
+			return
+		case value := <-watchCh:
+			if value == nil {
+				continue
+			}
+			switch value.Operation() {
+			case store.KVPut:
+				retries := 1
+				p := v.GetProcessor(value.Key())
+				for p == nil && retries <= 10 {
```

Comment:
what if we log as an Error and skip? did you see a case where you saw that the processor was not updated but got an OT event?

Reply:
> did you see a case where you saw that the processor was not updated but got an OT event?

So far only in my unit tests.

Reply:
let's use `ExponentialBackoff` with a cap? it will look cleaner.